src/ngraph_backend_manager.h

```diff
@@ -83,8 +125,17 @@ class BackendManager {
   // set of backends supported by nGraph
   static unordered_set<string> ng_supported_backends_;
 
+  // map of cached backend config objects
+  static unordered_map<string, std::shared_ptr<BackendConfig>>
+      ng_backendconfig_map_;
+  static mutex ng_backendconfig_map_mutex_;
+
   // Map of backends and their reference counts
   static std::map<std::string, int> ref_count_each_backend_;
+
+  // utility functions
+  static std::shared_ptr<BackendConfig> GetBackendConfig(
```

Comment:
on using unique_ptr, i get this error
`make_unique is not a member of ‘std’`
It seems our compiler doesn't support this
https://stackoverflow.com/questions/24609271/errormake-unique-is-not-a-member-of-std

Using shared_ptr for now

Reply:
You don't need to use the `make_unnique` API for creating a unique_ptr. Please don't use shared_ptr as this Ptr is not really shared. Use a unique ptr instead.

Reply:
That's correct. I did change it to create a unique_ptr without using make_unique. We have a map that holds onto the backend-config object ptr, and a utility function GetBackendConfig(string backend_name) that returns the ptr by querying the map. Was running into issues around it. Let me try again.